blog/2018-12-04-cache-invalidation-with-change-data-capture.adoc

```diff
@@ -0,0 +1,490 @@
+===============================
+
+Modelling order and item as JPA entities is straight-forward:
+
+[source,java]
+----
+@Entity
+public class PurchaseOrder {
+
+    @Id
+    @GeneratedValue(generator = "sequence")
+    @SequenceGenerator(
+        name = "sequence", sequenceName = "seq_po", initialValue = 1001, allocationSize = 50
+    )
+    private long id;
+    private String customer;
+    @ManyToOne private Item item;
+    private int quantity;
+    private BigDecimal totalPrice;
+
+    // ...
+}
+----
+
+As changes to items are rare, the `Item` entity should be cached.
+This can be done by simply specifying the `@Cacheable` annotation:
+
+[source,java]
+----
+@Entity
+@Cacheable
+public class Item {
+
+    @Id
+    private long id;
+    private String description;
+    private BigDecimal price;
+
+    // ...
+}
+----
+
+We also need to enable the second-level cache in the _META-INF/persistence.xml_ file.
+The property `hibernate.cache.use_second_level_cache` activates the cache itself, and the `ENABLE_SELECTIVE` cache mode
+causes only those entities to be put into the cache which are annotated with `@Cacheable`.
+It's also a good idea to enable SQL query logging and cache access statistics.
+That way we'll be able to verify whether things work as expected by examining the application log:
+
+[source,xml]
+----
+<?xml version="1.0" encoding="utf-8"?>
+<persistence xmlns="http://xmlns.jcp.org/xml/ns/persistence"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xsi:schemaLocation="..."
+    version="2.2">
+
+    <persistence-unit name="orders-PU-JTA" transaction-type="JTA">
+        <jta-data-source>java:jboss/datasources/OrderDS</jta-data-source>
+        <shared-cache-mode>ENABLE_SELECTIVE</shared-cache-mode>
+        <properties>
+            <property name="hibernate.cache.use_second_level_cache" value="true" />
+
+            <property name="hibernate.show_sql" value="true" />
+            <property name="hibernate.format_sql" value="true" />
+            <property name="hibernate.generate_statistics" value="true" />
+
+            <!-- dialect etc. ... -->
+        </properties>
+    </persistence-unit>
+</persistence>
+----
+
+When running on a Java EE application server, that's all you need to enable second-level caching.
+In the case of http://wildfly.org/[WildFly] (which is what's used in the example project), the http://infinispan.org/[Infinispan] key/value store is used as the cache provider by default.
+
+Now let's try and see what happens when modifying an item's price by running some SQL in the database,
+bypassing the application layer.
+If you've checked out the example source code, comment out the `DatabaseChangeEventListener` class and start the application as described in the _README.md_.
+You then can place purchase orders using curl like this
+(a couple of example items have been persisted at application start-up):
+
+[source,bash]
+----
+> curl -H "Content-Type: application/json" \
```

Comment:
Example repo uses httpie, blogpost uses curl. It might be good to unify it.

Reply:
Done.